dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/workflow/execution/WorkflowExecutionWorkerTest.java

```diff
@@ -0,0 +1,288 @@
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+
+class WorkflowExecutionWorkerTest extends WorkflowJMSTestBase implements WorkflowStepSPI {
+
+    static final String TEST_PROVIDER_ID = "test";
+    static final String SUCCESS_STEP_ID = "successful";
+    static final String PAUSING_STEP_ID = "pausing";
+    static final String FAILING_STEP_ID = "failing";
+
+    Clock clock = Clock.fixed(Instant.parse("2020-06-01T09:10:20.00Z"), UTC);
+
+    SettingsServiceBean settings = new TestSettingsServiceBean();
+    StubJpaPersistence persistence = new StubJpaPersistence();
+    DatasetRepository datasets = persistence.stub(DatasetRepository.class);
+    WorkflowRepository workflows = persistence.stub(WorkflowRepository.class);
+    WorkflowExecutionRepository executions = persistence.stub(WorkflowExecutionRepository.class);
+    WorkflowStepRegistry steps = new WorkflowStepRegistry() {{ init(); }};
+    RoleAssigneeServiceBean roleAssignees = new MockRoleAssigneeServiceBean() {{ add(new MockAuthenticatedUser()); }};
+    AuthenticationServiceBean authentication = new MockAuthenticationServiceBean(clock);
+    Instance<WorkflowExecutionListener> executionListeners = mock(Instance.class);
+
+    WorkflowExecutionContextFactory contextFactory = new WorkflowExecutionContextFactory(
+        settings, datasets, workflows, executions, roleAssignees, authentication);
+
+    WorkflowExecutionScheduler scheduler = new WorkflowExecutionScheduler() {{
+        setQueue(queue); setFactory(factory); }};
+
+    WorkflowExecutionServiceBean service = new WorkflowExecutionServiceBean(
+            datasets, executions, contextFactory, scheduler, clock);
+
+    WorkflowExecutionWorker worker = new WorkflowExecutionWorker(
+        datasets, steps, executions, contextFactory, scheduler, executionListeners, clock);
+
+    WorkflowExecutionWorkerTest() throws NamingException { }
+
+    @BeforeEach
+    void setUp() {
+        steps.register(TEST_PROVIDER_ID, this);
+
+        doNothing().when(datasets)
+                .lockDataset(any(Dataset.class), any(AuthenticatedUser.class), any(DatasetLock.Reason.class));
+        doNothing().when(datasets)
+                .unlockDataset(any(Dataset.class), any(DatasetLock.Reason.class));
+    }
+
+    @Test
+    void shouldExecuteSimpleWorkflowSuccessfully() throws Exception {
+        // given
+        Dataset dataset = datasets.save(givenDataset(clock));
+        Workflow workflow = workflows.save(givenWorkflow(1L,
+                givenWorkflowStep(TEST_PROVIDER_ID, SUCCESS_STEP_ID))
+        );
+        WorkflowContext context = givenWorkflowExecutionContext(dataset.getId(), workflow);
+        // when
+        givenMessageConsumer(worker)
+                .callProducer(() -> service.start(workflow, context))
+                .andAwaitMessages(2);
+        // then
+        List<WorkflowExecution> executions = persistence.stubRepositoryFor(WorkflowExecution.class).findAll();
```

Comment:
maybe if we refactor names for repositories - this can be obtained by executionRepository.findAll() ?